activesupport/lib/active_support/log_subscriber.rb

```diff
@@ -157,15 +159,32 @@ def #{level}(progname = nil, &block)
       METHOD
     end
 
-    # Set color by using a symbol or one of the defined constants. If a third
-    # option is set to +true+, it also adds bold to the string. This is based
-    # on the Highline implementation and will automatically append CLEAR to the
-    # end of the returned String.
-    def color(text, color, bold = false) # :doc:
+    # Set color by using a symbol or one of the defined constants. Set modes
+    # by specifying bold, italic, or underline options. Inspired by Highline,
+    # this method will automatically clear formatting at the end of the returned String.
+    def color(text, color, mode_options = {}) # :doc:
       return text unless colorize_logging
       color = self.class.const_get(color.upcase) if color.is_a?(Symbol)
-      bold  = bold ? BOLD : ""
-      "#{bold}#{color}#{text}#{CLEAR}"
+      mode = mode_from(mode_options)
+      "#{mode}#{color}#{text}\e[#{CLEAR}m"
+    end
+
+    def mode_from(options)
+      if options.is_a?(TrueClass) || options.is_a?(FalseClass)
+        ActiveSupport::Deprecation.warn(<<~MSG)
+          Bolding log subscriber text with a positional boolean is deprecated and will be removed in Rails 7.2.
+          Use an option hash instead: `color(:red, "my text", bold: true)`.
+        MSG
+        options = { bold: options }
+      end
+
+      codes = [].tap do |mode|
```

Comment:
I don't know if object allocations are a concern here, but you could default `mode_options = nil` and do an early `return if mode_options.nil?` at the top of this method. That would avoid allocating a useless array.

Reply:
I chose to make the modes a hash instead which simplified the process of building up the mode string. My motivation for using a hash is that I would like to use keywords instead when the deprecation is removed.